elixir/apps/web/lib/web/live/settings/identity_providers/okta/components.ex

```diff
@@ -0,0 +1,113 @@
+      <.form for={@form} phx-change={:change} phx-submit={:submit}>
+        <.step>
+          <:title>Step 1. Create a new App Integration in Okta</:title>
+          <:content>
+            <p class="mb-4">
+              Ensure the following scopes are added to the OAuth application:
+            </p>
+            <.code_block
+              id="oauth-scopes"
+              class="w-full text-xs mb-4 whitespace-pre-line rounded"
+              phx-no-format
+            ><%= scopes() %></.code_block>
+
+            <p class="mb-4">
+              Ensure the OAuth application has the following redirect URLs whitelisted:
+            </p>
+            <p class="mt-4">
+              <.code_block
+                :for={
+                  {type, redirect_url} <- [
+                    sign_in: url(~p"/#{@account.id}/sign_in/providers/#{@id}/handle_callback"),
+                    connect:
+                      url(~p"/#{@account.id}/settings/identity_providers/okta/#{@id}/handle_callback")
+                  ]
+                }
+                id={"redirect_url-#{type}"}
+                class="w-full mb-4 text-xs whitespace-nowrap rounded"
+                phx-no-format
+              ><%= redirect_url %></.code_block>
+            </p>
+          </:content>
+        </.step>
+
+        <.step>
+          <:title>Step 2. Configure Firezone</:title>
+          <:content>
+            <.base_error form={@form} field={:base} />
+
+            <div class="grid gap-4 mb-4 sm:grid-cols-1 sm:gap-6 sm:mb-6">
+              <div>
+                <.input
+                  label="Name"
+                  autocomplete="off"
+                  field={@form[:name]}
+                  placeholder="Name this identity provider"
+                  required
+                />
+                <p class="mt-2 text-xs text-neutral-500">
+                  A friendly name for this identity provider. This will be displayed to end-users.
+                </p>
+              </div>
+
+              <.inputs_for :let={adapter_config_form} field={@form[:adapter_config]}>
+                <div>
+                  <.input
+                    label="Client ID"
+                    autocomplete="off"
+                    field={adapter_config_form[:client_id]}
+                    required
+                  />
+                  <p class="mt-2 text-xs text-neutral-500">
+                    The Client ID from the previous step.
+                  </p>
+                </div>
+
+                <div>
+                  <.input
+                    label="Client secret"
+                    autocomplete="off"
+                    field={adapter_config_form[:client_secret]}
+                    required
+                  />
+                  <p class="mt-2 text-xs text-neutral-500">
+                    The Client secret from the previous step.
+                  </p>
+                </div>
+
+                <div>
+                  <.input
```

Comment:
Do we need an input for `api_base_url` setting?

Reply:
Good question.  I was back and forth on this one.  The Okta docs have the following:
```
.well-known/openid-configuration

Org Authorization Server
https://${yourOktaDomain}/.well-known/openid-configuration


Custom Authorization Server
https://${yourOktaDomain}/oauth2/${authorizationServerId}/.well-known/openid-configuration
```

and the "Core Okta API" docs have:
```
All URLs listed in the documentation should be preceded with your organization's subdomain (tenant) or configured custom domain.
```

With that in mind I decided that we should reliably be able to get a customers API Base URL from their `.well-known/openid-configuration` URL they provide to us, so I figured it'd be easier on people to not have to input another bit of info explicitly and it would also possibly prevent people from accidentally typing in the wrong API Base URL since we validate the `.well-known/openid-configuration` URL before they submit.

Curious on your thoughts though.

Reply:
Should we just ask for `yourOktaDomain` and build those URLs from it? I think it might be easier to find that rather than constructing `.well-known` by yourself

Reply:
So, the only downside I see with that is if they are using an "Okta Authorization Server", because then they will need to also add their `authorizationServerId`, so that would be another field too.  And if they aren't using an Authorization Server then they would have an empty field in their form

Reply:
Yeah, I think we hit this issue in 0.7 IIRC and settled on just allowing them to provide the whole discovery document URI, but with a nice placeholder / docs showing them how to get it.

Reply:
@AndrewDryga, I've updated the PR to use the OAuth Authorization Server URI as we talked about offline.  Let me know if you see any issues with it.